system/security/shared/caching.cpp

```diff
@@ -672,16 +681,39 @@ bool CPermissionsCache::queryPermsManagedFileScope(ISecUser& sec_user, const cha
 
 SecAccessFlags CPermissionsCache::queryDefaultPermission(ISecUser& user)
 {
-    if (m_defaultPermission == SecAccess_Unknown)
+    if (!m_secMgr)
+        return SecAccess_Full; // if no security manager, all full access to all scopes
+
+    if (m_useLegacyDefaultFileScopePermissionCache)
     {
-        if (m_secMgr)
+        if (m_defaultPermission == SecAccess_Unknown)
+        {
             m_defaultPermission = m_secMgr->queryDefaultPermission(user);
-        else
-            m_defaultPermission = SecAccess_None;
+            DBGLOG("Legacy default file scope permission set to %s(%d) for all users, based on User '%s'", getSecAccessFlagName(m_defaultPermission),
+                   m_defaultPermission, user.getName());
+        }
+        return m_defaultPermission;
     }
-    return m_defaultPermission;
 
+    SecAccessFlags defaultPermission = SecAccess_None;
+    CriticalBlock defaultScopePermissionBlock(syncDefaultScopePermissions);
+    const std::string username(user.getName());
```

Comment:
minor: This code doesn't need to be inside the critical section.  Minimising the code with a mutex helps reduce contention.  I will merge as-is.